VideoRenderer/VideoRenderer/SphereVideoStreamViewController.swift

```diff
@@ -113,39 +101,39 @@ public class SphereVideoStreamViewController: GLKViewController, RendererProtoco
                 
                 currentPlayer = AVPlayer(url: props.content)
                 
-                weak var this = self
-                var callbacks = PlayerObserver.Callbacks()
-                
-                callbacks.failedState = {
-                    this?.dispatch?(.playbackFailed($0))
-                }
-                
-                callbacks.rateChanged = { new, old in
-                    guard new != old else { return }
-                    if new == 0 { this?.dispatch?(.playbackStopped) }
-                    else { this?.dispatch?(.playbackStarted) }
-                }
-                
-                callbacks.durationDefined = { duration, _ in
-                    this?.dispatch?(.durationReceived(duration))
-                }
-                
-                callbacks.endOfVideo = {
-                    this?.dispatch?(.playbackFinished)
-                }
-                
-                callbacks.playbackReady = {
-                    guard $0 == true else { return }
-                    this?.dispatch?(.playbackReady)
-                }
-                
-                callbacks.bufferedTimeUpdated = {
-                    this?.dispatch?(.bufferedTimeUpdated($0))
+                observer = SystemPlayerObserver(player: currentPlayer) { [weak self] event in
+                    switch event {
+                    case .didChangeItemStatus(_, let new):
```

Comment:
We never use `.old` value of duration from `.didChangeItemStatus` and probably never will, we can drop it observation.